### ISSUE TITLE
File removing in mounted .tar archive not working

### ISSUE THREAD

**Issue by r3t4k3r** [mxmlnkn/ratarmount#133]
Adding new file/edit work fine. But remove file still not working. No any errors shown.

![image](https://github.com/mxmlnkn/ratarmount/assets/86055096/6fe8eece-d7ee-4d34-9848-ecb0f3d2a8d2)

**Comment by mxmlnkn** [comment]
Hi, nice lain avatar. I'm sorry for your troubles. Because of the other write overlay issues, I am planning to redo the TAR commit feature in the future. Currently, as you can see, it simply calls the installed GNU tar command to delete files from and append files to the TAR archive. You can copy-paste the shown commands in a shell to do the committing manually. That way, you can also debug it. For example:

 - What are the contents in `deletion.lst`? While the confirmation prompt is shown and not yet confirmed, the file should exist inside the displayed `/tmp` subfolder. Is the file to be deleted at least listed in that `deletion.lst`? If so, it might also be a good idea to run the tar command without the `2>&1 | sed ...`, which is there to filter warnings.
 - The output of `tar tvlf <file.tar>` would also be helpful. For example, files inside the TAR may have some denormal path names such as `././file`, in which case the deletion list may fail to match resulting in the file not being deleted.

Alternatively, it would be very helpful if you could send me a minimal TAR file and bash script to reproduce the bug so that I can debug the issue myself.

**Comment by r3t4k3r** [comment]
### Repeat steps:

#### create test dir
```
mkdir test
cd test
```

#### create test files
```
touch 1.txt
touch 2.txt
```

#### create archive
```
tar -cf foo.tar 1.txt 2.txt
```

#### create folders for mount
```
mkdir foo.overlay
mkdir foo
```


#### 
`ls` (Optional)
```
1.txt  2.txt  foo/  foo.overlay/  foo.tar
```

#### mount
```
ratarmount -o allow_other --write-overlay "foo.overlay" "foo.tar" "foo"
```

#### foo content
`ls -la foo` (Optional)
```
total 2
drwxrwxrwx 1 noanon noanon  0 Apr 30 01:38 ./
drwxr-xr-x 1 noanon noanon 62 Apr 30 01:37 ../
-rw-r--r-- 1 noanon noanon  0 Apr 30 01:37 1.txt
-rw-r--r-- 1 noanon noanon  0 Apr 30 01:37 2.txt
```

#### remove test file
```
rm foo/1.txt
```

#### check is file deleted
`ls -la foo` (Optional)
```
total 1
drwxrwxrwx 1 noanon noanon  0 Apr 30 01:38 ./
drwxr-xr-x 1 noanon noanon 62 Apr 30 01:37 ../
-rw-r--r-- 1 noanon noanon  0 Apr 30 01:37 2.txt
```

#### unmout fuse archive
```
fusermount -u foo
```

#### apply overlay changes
```
ratarmount --commit-overlay --write-overlay "foo.overlay" "foo.tar" "foo"
```
Optional check `/tmp/tmp*` content
![image](https://github.com/mxmlnkn/ratarmount/assets/86055096/95d6967c-25cb-414d-a102-70e6a1de3915)

write `commit` and press `Enter` key

#### Check output archive content
```
tar tvlf foo.tar
```
Output:
```
-rw-r--r-- noanon/noanon     0 2024-04-30 01:37 1.txt
-rw-r--r-- noanon/noanon     0 2024-04-30 01:37 2.txt
```

How we can see, file not deleted
![image](https://github.com/mxmlnkn/ratarmount/assets/86055096/0faa097d-b6e8-4d28-9fac-a4afcd209635)

**Comment by r3t4k3r** [comment]
I do how you said, and delete pipe. This is error while delete command
![image](https://github.com/mxmlnkn/ratarmount/assets/86055096/1b559fe7-257c-4fa4-8c46-114882e6cf34)

**Comment by mxmlnkn** [comment]
Thank you. I can reproduce it with your scripts. The deletion list isn't created correctly. I fixed it in 76820c433e8b993edabda2de5c3a6f5e757d4543. You can try it out with:

```bash
python3 -m pip install --user --force-reinstall \
    'git+https://github.com/mxmlnkn/ratarmount.git@master#egginfo=ratarmount'
```
